tests/helper/helper_generic.go

```diff
@@ -330,6 +333,7 @@ func CommonBeforeEach() CommonVar {
 func CommonAfterEach(commonVar CommonVar) {
 	// Get details, including test filename, test case name, test result, and test duration for each test spec and adds it to local testResults.txt file
 	// Ginkgo test related variables
+	CreateTelemetryDebugFile()
```

Comment:
Why do you have this here and in CommonBeforeEach?

Reply:
ok removing it

Reply:
I think this whole file can be left as it is. We don't need to make any change to this file, any change required can be made in the `BeforeEach` and `AfterEach` block of tests.
```go
	When("recording telemetry data", func() {
		BeforeEach(func() {
			helper.CreateTelemetryDebugFile()
			helper.Cmd("odo"," <command>").ShouldPass().Out()
		})
		AfterEach(func() {
			helper.ResetTelemetry()
		})
		It("should record the telemetry data correctly", func() {
			td := helper.GetTelemetryDebugData()
			Expect(td.Event).To(ContainSubstring("odo <command>"))
			Expect(td.Properties.Success).To(BeTrue())
			Expect(td.Properties.Error == "").To(BeTrue())
			Expect(td.Properties.ErrorType == "").To(BeTrue())
                        ....
                        ....
		})
	})
```

Reply:
Removed